packages/api/src/mcp/parsers.ts

```diff
@@ -183,7 +183,13 @@ export function formatToolContent(
   }
 
   if (uiResources.length) {
-    formattedContent.push({ type: 'text', metadata: 'ui_resources', text: btoa(JSON.stringify(uiResources))});
+    formattedContent.push({
+      type: 'text',
+      metadata: {
+        type: 'ui_resources',
+        data: uiResources
```

Comment:
We now pass the `uiResources` array directly in the metadata. This way, we don't need to do any escaping/unescaping/b64encoding/decoding. Much simpler and robust!